File: src/tui/snippets/sample.js
// Syntax preview: comments, keywords, types, strings, escapes, labels.

const MAX_ITEMS = 100;
const VERSION = "1.0.0";

class Config {
    constructor(name) {
        this.name = name;
        this.count = 0;
        this.enabled = true;
    }
    validate() {
        if (!this.name) throw new Error("Name cannot be empty");
        return true;
    }
}

function process(items) {
    const result = {};
    outer: for (const item of items) {
        if (item < 0) continue outer;
        result[item] = item % 2 === 0;
    }
    return result;
}

function parseEmail(text) {
    const pattern = /[\w.-]+@[\w.-]+\.\w+/;
    const match = text.match(pattern);
    return match ? match[0] : null;
}

const msg = "Hello\tWorld\n";
const config = new Config("example");
config.validate();
console.log(`Config: ${JSON.stringify(config)}, msg: ${msg}`);
console.log(`Email: ${parseEmail("test@example.com")}`);
console.log(`Result: ${JSON.stringify(process([1, 2, -3, 4, 5]))}`);

File: src/tui/snippets/sample.py
# Syntax preview: comments, keywords, types, strings, escapes, regex.

import re
from dataclasses import dataclass
from typing import Dict, List, Optional

MAX_ITEMS = 100
VERSION = "1.0.0"

@dataclass
class Config:
    name: str
    count: int = 0
    enabled: bool = True

    def validate(self) -> Optional[str]:
        if not self.name:
            return "Name cannot be empty"
        return None

def process(items: List[int]) -> Dict[int, bool]:
    result = {}
    for item in items:
        if item < 0:
            continue
        result[item] = item % 2 == 0
    return result

def parse_email(text: str) -> Optional[str]:
    pattern = r"[\w.-]+@[\w.-]+\.\w+"
    match = re.search(pattern, text)
    return match.group(0) if match else None

if __name__ == "__main__":
    msg = "Hello\tWorld\n"
    config = Config(name="example")
    print(f"Config: {config}, msg: {msg!r}")
    print(f"Email: {parse_email('test@example.com')}")
    print(f"Result: {process([1, 2, -3, 4, 5])}")
